microsoft/testsuites/core/azure_image_standard.py

```diff
@@ -901,7 +901,11 @@ def verify_no_pre_exist_users(self, node: Node) -> None:
             r" \\\"root\\\".\';echo;sleep .*\"",
             re.M,
         )
-        current_user = node.tools[Whoami].get_username()
+        # For Bitnami images, the bitnami service will add changes for the current admin
+        # user. The output of the command "whoami" is not the logged user. So we get the
+        # username from node connetion info.
+        remote_node = cast(RemoteNode, node)
+        current_user = remote_node._connection_info.username
```

Comment:
Please double check the mechanism, make sure this is the right way.

Reply:
Yes, remote_node._connection_info.username is the admin user that is set in the arm template. I have run it on about 1500 images. That works well.

Reply:
For most images, the name detected by the system is the same as the original name. For now, only the name of bitnami image detected by the system is another name "bitnami" instead of "lisatest". Following the logic of this case, the "current_user" should be "lisatest" not "bitnami". So checking by the original name is expected.

This test case aims to detect if the authorized_keys files or passwords of previous users existed in the image. It needs to exclude the user ("lisatest" by default) which Lisa adds for the connection. (This user is the original admin user that sets in the arm template. After the VM boot up, this user and its authorized_keys are added by cloud-init.)

Before this fix, for bitnami images,  following the logic of this case,  "lisatest" is considered as the pre-existing user.  authorized_keys file is detected under /home/lisatest/.ssh/, then this case reports a false failed with the message "/home/lisatest/.ssh/authorized_keys is detected". 

PS:
Why the output of "whoami" is "bitnami" rather than "lisatest"? The bitnami.service changes the UID of "lisatest" as the UID 1000 which the name "bitnami" already has. "whoami" gets the name according to the UID, then the output is "bitnami".

Reply:
Thanks for detail explanation!